src/main/kotlin/com/matthewrussell/trchunkbrowser/app/mainview/MainView.kt

```diff
@@ -25,9 +25,29 @@ class MainView : View() {
     override val root = stackpane {
         vbox {
             addClass(MainViewStyles.root)
-            add(JFXSnackbar(this).apply {
+            add(JFXSnackbar(this).apply { // Progress snackbar
+                viewModel.snackBarProgress.subscribe { progress ->
+                    if(progress < 100) {
+                        enqueue(JFXSnackbar.SnackbarEvent(
+                            messages["export_in_progress"],
+                            "",
+                            0,
+                            true,
+                            null))
+                    } else {
+                        // wait until snackbar is shown to close it
+                        while (!this.isVisible) {
+                            Thread.sleep(200)
```

Comment:
Instead of doing this, can you use visibleProperty().onChange { }  ?

Reply:
@mXaln this might work...

```
this.visibleProperty().toObservable().takeWhile { !it }.subscribe {
     if(it) this.close()
}
```

Or...
```
this.visibleProperty().toObservable().takeWhile { !it }.doOnComplete {
    this.close()
}
```